### PR TITLE
Adding appnexus debug via cookie/params

### PR DESCRIPTION
**Pull request by aneuway2**
## Type of change
- [x] Feature

## Description of change
The following feature enhancement adds in the ability for a publisher or developer to request an AppNexus AST debug auction in the response back from AppNexus alongside any bids.

This feature can be enabled by either of the following two methods along with the `pbjs_debug=true` query string parameter.

1) By adding a `debug` object in parallel to the bids requested:

```
{
  bidder: 'appnexus',
  params: {
    placementId: 123456789
  },
  debug: {
    enabled: true,
    dongle: 'QWERTY',
    member_id: 958,
    debug_timeout: 3000
  }
}
```

2) By setting an `apn_prebid_debug` cookie with a JSON string:

```
{ "enabled": true, "dongle": "QWERTY", "debug_timeout": 1000, "member_id": 958 }
```

When both of the above are enabled, settings from the browser cookie will prevail.

There also is no guarantee that the prebid debug may not freeze the browser (e.g in the case where gigabytes of data are returned) but this should be okay since this enables further debug information for developers only.

For more information on how to read the debug auction results, please see https://wiki.appnexus.com/display/console/Debug+Auction+Glossary

*Sample Console Log:*

```
Running member 958 debug using no permissions
Impbus 4280.bm-impbus.prod.nym2:8001
	Header Referer: http://example.com?pbjs_debug=true&ast_debug=true&ast_dongle=sss
	Site Domain: example.com
1772142023491463321 - Predicted engagement_rate_id: 2 (view) imps 54.87%
1772142023491463321 - Predicted engagement_rate_id: 3 (view_over_total) imps 51.57%
1772142023491463321 - Predicted engagement_rate_id: 8 (predicted_100pv1s_display_view_rate) imps 47.06%
```

### CODE REVIEW

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging d633c7bd7f47ac05685eb23ace22d3d2dd40636c into e7ad006f2f1c14978e0175fbdb935e5ca1dca547 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-f3068f58667865de10eb045351c9237c44c8a404)

**new alerts:**

* 1 for Useless conditional

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by aneuway2** [comment]
@jsnellbaker - made changes to make that more straightforward.

> if someone misconfigured one of the variables
This should be okay. The debug auction will still be returned as long as enabled=true. I did add a check to make sure this was true

> what docs changes are you thinking for this addition
I was planning on adding this as a separate section at the bottom of the readme on prebid.org with pretty much the same language as what is in this PR description

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging d51c8980c96f682cfecf2014ccb157807296fe81 into c2847abc0dfb69667dd4365388a7bf127f1c73fa - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-d0709d1dda4f4499589b05402cae85ec07cf1b17)

**new alerts:**

* 1 for Useless conditional

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by mkendall07** [comment]
This pull request **introduces 1 alert** when merging d51c8980c96f682cfecf2014ccb157807296fe81 into 3ba867ae7157c00c6414379a3629e95b00b102e9 - [view on LGTM.com](https://lgtm.com/projects/g/prebid/Prebid.js/rev/pr-e28d7e3e6d942225860b05b8272c07a8defdf7d9)

**new alerts:**

* 1 for Useless conditional

---

*Comment posted by [LGTM.com](https://lgtm.com)*

**Comment by jsnellbaker** [comment]
@aneuway2 Thanks for making the changes to the code; it looks much better now.  Would you be able to quickly address the LGTM warning/alert?

Also, did you get the chance to put together the docs PR?  Once that's made/linked here, we should be good to merge.

File: modules/appnexusBidAdapter.js [hunk 1]
```diff
@@ -198,6 +198,7 @@ export const spec = {
         .replace(/<h1>(.*)<\/h1>/gm, '\n\n===== $1 =====\n\n') // Header H1
         .replace(/<h[2-6]>(.*)<\/h[2-6]>/gm, '\n\n*** $1 ***\n\n') // Headers
         .replace(/(<([^>]+)>)/igm, ''); // Remove any other tags
+      utils.logMessage("AppNexus Debug Auction Glossary: https://wiki.appnexus.com/x/qwmHAg");
```
Comment (jsnellbaker) [inline modules/appnexusBidAdapter.js]: Can you change the double-quotes to single-quotes?  It's causing a linting error.

Comment (aneuway2) [reply modules/appnexusBidAdapter.js]: yup. of course :-) Just pushed a fix.

**Comment by aneuway2** [comment]
Added docs PR @jsnellbaker 👍

**Pull request merged**